### PR TITLE
Homework #6 is completed

### PR DESCRIPTION
**Pull request by komyagin**

### CODE REVIEW

File: src/main/java/hw6/pages/enteties/JdiUser.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package hw6.pages.enteties;
+
+import lombok.AllArgsConstructor;
+import lombok.Setter;
+
+import java.util.Objects;
+
+@Setter
+@AllArgsConstructor
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw6/pages/enteties/JdiUser.java:8]: Why do you implement when you use `lombok`
* `toString`
* `equals` 
* `hashCode`

File: src/test/java/hw6/exercise1/SoftAssertWebElementsTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import hw6.BaseTest;
+import hw6.pages.JdiTestingHomePage;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+public class SoftAssertWebElementsTest extends BaseTest {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/exercise1/SoftAssertWebElementsTest.java]: Why do you need this test in the current hw?

File: src/test/java/hw6/exercise2/AssertWebElementsOnPageTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+
+public class AssertWebElementsOnPageTest extends BaseTest {
+
+    @Test
+    public void HardAssertWebElements() {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/exercise2/AssertWebElementsOnPageTest.java]: Why do you need this test in the current hw?

File: src/test/java/hw6/steps/GivenSteps.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Given("I go through the header menu Service -> Different Elements Page")
+    public void iOpenDifferentElementsPageThroughHeaderOnIndexPage() {
+        homePage.goToDifferentElementsPage();
+    }
+
+    @Given("I go through the header menu Service -> User Table")
+    public void iOpenUserTablePageThroughHeaderOnIndexPage() {
+        homePage.goToUserTablePage();
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/GivenSteps.java]: What is the difference betwee these two steps?

File: src/test/java/hw6/steps/GivenSteps.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @Given("I am on Home page")
+    public void iAmOnHomePage() {
+        WebDriverSingleton.INSTANCE.getDriver().get(Properties.getProperty("url"));
+        homePage = new JdiTestingHomePage(WebDriverSingleton.INSTANCE.getDriver());
+    }
+
+    @Given("I login as username: {string} password: {string}")
+    public void iLoginOnIndexPage(String username, String password) {
+        homePage.login(username, password);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/GivenSteps.java]: I saw `User` object in your code. Why did you do not use here?

File: src/test/java/hw6/steps/ThenSteps.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Then("User table should contain following values:")
+    public void iAssertContentOfTableOnUserTablePAge(List<JdiUser> users) {
+        List<JdiUser> actualUserTable = new ArrayList<>();
+        List<String> numbers = userTablePage.getUsersIds();
+        List<String> usernames = userTablePage.getUsersNames();
+        List<String> description = userTablePage.getUsersImagesDescriptions();
+        for (int i = 0; i < numbers.size(); i++) {
+            actualUserTable.add(new JdiUser(numbers.get(i), usernames.get(i), description.get(i)));
+        }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw6/steps/ThenSteps.java]: May be it is better incapsulate this user created to the Page object?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please resolve comments

File: src/main/java/hw6/pages/enteties/JdiUser.java (hunk 1 above)
Comment (komyagin) [reply src/main/java/hw6/pages/enteties/JdiUser.java:8]: I didn't know that Lombok has a EqualsAndHashCode annotations. Changed

File: src/test/java/hw6/exercise1/SoftAssertWebElementsTest.java (hunk 2 above)
Comment (komyagin) [reply src/test/java/hw6/exercise1/SoftAssertWebElementsTest.java]: Deleted

File: src/test/java/hw6/exercise2/AssertWebElementsOnPageTest.java (hunk 3 above)
Comment (komyagin) [reply src/test/java/hw6/exercise2/AssertWebElementsOnPageTest.java]: Deleted

File: src/test/java/hw6/steps/GivenSteps.java (hunk 4 above)
Comment (komyagin) [reply src/test/java/hw6/steps/GivenSteps.java]: Changed to switch method

File: src/test/java/hw6/steps/GivenSteps.java (hunk 5 above)
Comment (komyagin) [reply src/test/java/hw6/steps/GivenSteps.java]: Changed

File: src/test/java/hw6/steps/ThenSteps.java (hunk 6 above)
Comment (komyagin) [reply src/test/java/hw6/steps/ThenSteps.java]: Done

**Review by DmitryKhodakovsky — APPROVED** [review]
Approved

**Pull request merged**